frontend/src/components/@shared/SearchBar/SearchBar.tsx

```diff
@@ -1,14 +1,70 @@
-import { InputHTMLAttributes } from 'react';
-import { Container } from './SearchBar.styles';
+import { InputHTMLAttributes, ChangeEvent, useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+
+import SearchIcon from 'src/components/Icons/search';
+import { COLOR, PATH } from 'src/constants';
+import Arrow from '../Arrow/Arrow';
+import { Container, ResetButton, SearchButton, SearchInput } from './SearchBar.styles';
 
 const SearchBar = ({
   onSubmit,
+  onClick,
   placeholder,
 }: InputHTMLAttributes<HTMLInputElement | HTMLFormElement>) => {
+  const history = useHistory();
+  const location = useLocation();
+
+  const [isMainPage, setIsMainPage] = useState(true);
```

Comment:
이건 현재 history와 location을 사용하고 있어서

```suggestion
  const isMainPage = location.pathname === (PATH.HOME || PATH.ROOT)
```

처럼 사용해도 괜찮을 것 같아요. 상태로 관리될 필요는 없어 보입니다.

Reply:
그렇네용! 
리액트에서는 거의 반 자동적으로 변수 선언을 할 때 그냥 상태로 관리하는데, 앞으론 좀더 고려해보고 상태/상태로 관리할 필요가 없는 것을 분리하겠습니당👍👍👍 https://github.com/woowacourse-teams/2021-jujeol-jujeol/pull/189/commits/3cb2f34ec1956673723153654a8ed595f7bdd1a0